### DISCUSSION TITLE
Error: "Failure obtaining db row lock: Execution Timeout Expired"

### DISCUSSION THREAD

**Discussion by vwitteki** [quartznet/quartznet#1437] [Q&A]
Hello!  We are using quartznet v3.0.7  in clustered mode in a .NET Framework 4.7.2 application, using the following settings:
```
{ "quartz.threadPool.type", "Quartz.Simpl.SimpleThreadPool, Quartz" },
{ "quartz.threadPool.threadCount", "10" },
{ "quartz.threadPool.threadPriority", "Normal" },
{ "quartz.jobStore.misfireThreshold", "60000" },
{ "quartz.jobStore.type", "Quartz.Impl.AdoJobStore.JobStoreTX, Quartz" },
{ "quartz.jobStore.useProperties",  "false" },
{ "quartz.jobStore.clustered", "true" },
{ "quartz.jobStore.acquireTriggersWithinLock", "false" },
{ "quartz.jobStore.dataSource", "default" },
{ "quartz.jobStore.tablePrefix", "QRTZ_" },
{ "quartz.jobStore.driverDelegateType", "Quartz.Impl.AdoJobStore.SqlServerDelegate, Quartz" },
{ "quartz.dataSource.default.connectionString", "[redacted]" },
{ "quartz.dataSource.default.provider", "[redacted]" },
{ "quartz.serializer.type", "json" },
{ "quartz.scheduler.instanceName", "QuartzScheduler" },
{ "quartz.scheduler.instanceId", "AUTO" },
```

After starting, it ran successfully for a little over 2 hours, and then we started to see the following errors:
```
Failure obtaining db row lock: Execution Timeout Expired. The timeout period elapsed prior to completion of the operation or the server is not responding.
Quartz.Impl.AdoJobStore.LockException: Failure obtaining db row lock: Execution Timeout Expired. The timeout period elapsed prior to completion of the operation or the server is not responding.
```

Digging into the SQL server traffic, it looks like these were caused by a long blocking chain on SELECT QUARTZ_LOCKS calls, which formed without an obvious explanation.  I don't think the load was the problem, because the application did not see an increase in load before failure, and also because it survived a test with an even heavier request load.

Is there a known issue in our quartznet version, or a known use condition that could cause something like this? Any recommendations for things to try in debugging or solving? Currently we are not able to reproduce the issue in test environments.

**Comment by lahma** [comment]
Usually in these cases I would blame backups or something running at the same time blocking, who knows. Your configuration seems otherwise fine.

**Reply by vwitteki** [reply]
Looked into it and found that no backups were running at this time. Do you have any other suggestions for things to look into? Does the issue seem like something that might be fixed in a newer quartznet version?

**Reply by lahma** [reply]
Well I hope that every version of Quartz is better than the one before. But you might need to also dig into SQL Server logs / traces to find out what caused the timeout. Generally if you have the basic setup with indexes in place, it should perform pretty well.
